operators/pkg/tenant-controller/tenant_controller.go

```diff
@@ -307,6 +347,72 @@ func (r *TenantReconciler) checkValidWorkspaces(ctx context.Context, tn *crownla
 	return tenantExistingWorkspaces, workspaces, err
 }
 
+// deleteClusterNamespace deletes the namespace for the tenant, if it fails then it returns an error
+func (r *TenantReconciler) deleteClusterNamespace(ctx context.Context, tn *crownlabsv1alpha2.Tenant, nsName string) (nsOk bool, err error) {
+	ns := v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: nsName}}
+
+	nsErr := utils.EnforceObjectAbsence(ctx, r.Client, &ns, "personal namespace");
+    
+	if nsErr != nil {
+		klog.Errorf("Error when deleting namespace of tenant %s -> %s", tn.Name, nsErr)
+		return false, nsErr
+	}
+
+	r.updateTnNamespace(&ns, tn.Name)
+	return true, nsErr
+}
+
+
+func (r *TenantReconciler) deleteCreateOrUpdateClusterResources(ctx context.Context, tn *crownlabsv1alpha2.Tenant, nsName string, tenantExistingWorkspaces []crownlabsv1alpha2.TenantWorkspaceEntry, workspaces []crownlabsv1alpha1.Workspace, keepNsOpen bool, retrigErr error) (nsOk bool, err error) {
+
+    // Delete namespace or update the cluster resources
+
+    nsOk = false; // nsOk must be initialized for later use
+
+	// update resource quota in the status of the tenant after checking validity of workspaces.
+	tn.Status.Quota = forge.TenantResourceList(workspaces, tn.Spec.Quota)
```

Comment:
This is unrelated from the namespace creation/deletion logic.

Reply:
Resolved in [89e5461](https://github.com/netgroup-polito/CrownLabs/pull/821/commits/89e5461399b979c4fd563130545d8ceec41cf91c)